src/main/kotlin/no/liflig/documentstore/dao/Transactional.kt

```diff
@@ -16,30 +16,25 @@ internal class CoroutineTransaction(
   override fun toString(): String = "CoroutineTransaction(handle=$handle)"
 }
 
-@Deprecated("Use transactional with jdbi and explicit handle instead")
-suspend fun <T> transactional(dao: CrudDao<*, *>, block: suspend () -> T): T = when (dao) {
-  is CrudDaoJdbi -> {
-    transactional(dao.jdbi) {
-      block()
-    }
-  }
-
-  else -> throw Error("Transactional requires JDBIDao")
-}
-
-suspend fun <T> transactional(jdbi: Jdbi, block: suspend (Handle) -> T): T {
+suspend fun <T> transactional(limiter: CoroutineJdbiWrapper, block: suspend (Handle) -> T): T {
   val existingHandle = coroutineContext[CoroutineTransaction]?.handle
 
   // We can assume that we're already in a transaction, so we just pass the existing handle into the block
   return if (existingHandle != null) {
     block(existingHandle)
   } else
     mapExceptions {
-      jdbi.open().use { handle ->
-        withContext(Dispatchers.IO + CoroutineTransaction(handle)) {
-          jdbi.transactionHandler.begin(handle)
-          block(handle)
-            .also { jdbi.transactionHandler.commit(handle) }
+      limiter.withHandle { handle ->
+        withContext(CoroutineTransaction(handle)) {
```

Comment:
Burde denne være
```withContext(CoroutineTransaction(handle) + coroutineContext) {```
?

Reply:
Jeg mener at withContext alltid legger til på den eksisterende, så det der skjer "behind the scenes" fra før

https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/with-context.html

> The resulting context for the [block](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/with-context.html) is derived by merging the current [coroutineContext](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.coroutines/index.html) with the specified [context](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/with-context.html) using coroutineContext + context (see [CoroutineContext.plus](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.coroutines/-coroutine-context/plus.html)). This suspending function is cancellable. It immediately checks for cancellation of the resulting context and throws [CancellationException](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-cancellation-exception/index.html) if it is not [active](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/is-active.html).

Reply:
Åja, så alle andre steder hvor vi har + coroutineContext er det bortkasta